devicetypes/smartthings/viconics-schneider-room-controller.src/viconics-schneider-room-controller.groovy

```diff
@@ -0,0 +1,532 @@
+
+	def child = addChildDevice("Child Thermostat Setpoints", "${device.deviceNetworkId}:1", device.hubId,
+			[completedSetup: true, label: childName, isComponent: true, componentName: "childSetpoints", componentLabel: label]
+	)
+
+	child.sendEvent(name: "coolingSetpoint", value: 20.0, unit: "C")
+	child.sendEvent(name: "heatingSetpoint", value: 21.0, unit: "C")
+	log.debug "child.inspect() ${child}"
+}
+
+def parse(String description) {
+    def result = []
+    def eventMap = [:]
+    def descMap = zigbee.parseDescriptionAsMap(description)
+
+    if (descMap.clusterInt == THERMOSTAT_CLUSTER && descMap.attrId) {
+        def attributeInt = zigbee.convertHexToInt(descMap.attrId)
+
+        switch (attributeInt) {
+            case OCCUPANCY:
+            case CUSTOM_EFFECTIVE_OCCUPANCY:
+                log.debug "${attributeInt == OCCUPANCY ? "OCCUPANCY" : "EFFECTIVE OCCUPANCY"}, descMap.value: ${descMap.value}, attrId: ${attributeInt}"
+                eventMap.name = "occupancy"
+                eventMap.value = EFFECTIVE_OCCUPANCY_MAP[descMap.value]
+                break
+            case COOLING_SETPOINT:
+                log.debug "COOLING SETPOINT OCCUPIED, descMap.value: ${descMap.value}"
+                eventMap.name = "coolingSetpoint"
+                eventMap.value = getTemperature(descMap.value)
+                eventMap.unit = temperatureScale
+                break
+            case HEATING_SETPOINT:
+                log.debug "HEATING SETPOINT OCCUPIED, descMap.value: ${descMap.value}"
+                eventMap.name = "heatingSetpoint"
+                eventMap.value = getTemperature(descMap.value)
+                eventMap.unit = temperatureScale
+                break
+            case COOLING_SETPOINT_UNOCCUPIED:
+                log.debug "COOLING SETPOINT UNOCCUPIED, descMap.value: ${descMap.value}"
+                def childEvent = [:]
+                childEvent.name = "coolingSetpoint"
+                childEvent.value = getTemperature(descMap.value)
+                childEvent.unit = temperatureScale
+                sendEventToChild(UNOCCUPIED_SETPOINT_CHILD_DEVICE_ID, childEvent)
+                break
+            case HEATING_SETPOINT_UNOCCUPIED:
+                log.debug "HEATING SETPOINT UNOCCUPIED, descMap.value: ${descMap.value}"
+                def childEvent = [:]
+                childEvent.name = "heatingSetpoint"
+                childEvent.value = getTemperature(descMap.value)
+                childEvent.unit = temperatureScale
+                sendEventToChild(UNOCCUPIED_SETPOINT_CHILD_DEVICE_ID, childEvent)
+                break
+            case LOCAL_TEMPERATURE:
+                log.debug "LOCAL TEMPERATURE, descMap.value: ${descMap.value}"
+                eventMap.name = "temperature"
+                eventMap.value = getTemperature(descMap.value)
+                eventMap.unit = temperatureScale
+                break
+            case CUSTOM_HUMIDITY:
```

Comment:
just double-checking, they included this under the thermostat cluster, right?

Reply:
@greens 
It's possible to obtain humidity values either from RELATIVE_HUMIDITY_CLUSTER or THERMOSTAT_CLUSTER (custom attribute)